TerraformCLI/task.json

```
@@ -1,151 +1,150 @@
-  "description": "Execute terraform cli commands",
-  "author": "Charles Zipp",
-  "helpMarkDown": "",
-  "category": "Utility",
-  "visibility": [
-    "Build",
-    "Release"
-  ],
-  "demands": [],
-  "version": {
-    "Major": "0",
-    "Minor": "2",
-    "Patch": "4"
-  },
-  "minimumAgentVersion": "1.95.0",
-  "instanceNameFormat": "terraform $(command)",
-  "groups" : [
-    {
-      "name": "backendAzureRm",
-      "displayName": "AzureRM Backend Configuration",
-      "isExpanded": false,
-      "visibleRule": "command = init && backendType = azurerm"
+    "id": "721c3f90-d938-11e8-9d92-09d7594721b5",
+    "name": "TerraformCLI",
+    "friendlyName": "Terraform CLI",
+    "description": "Execute terraform cli commands",
+    "author": "Charles Zipp",
+    "helpMarkDown": "",
+    "category": "Utility",
+    "visibility": [
+        "Build",
+        "Release"
+    ],
+    "demands": [],
+    "version": {
+        "Major": "0",
+        "Minor": "2",
+        "Patch": "4"
     },
-    {
-      "name": "variables",
-      "displayName": "Variables",
-      "isExpanded": false,
-      "visibleRule": "command = validate || command = plan || command = apply"
+    "minimumAgentVersion": "1.95.0",
+    "instanceNameFormat": "terraform $(command)",
+    "groups": [{
+            "name": "backendAzureRm",
+            "displayName": "AzureRM Backend Configuration",
+            "isExpanded": false,
+            "visibleRule": "command = init && backendType = azurerm"
+        },
+        {
+            "name": "variables",
+            "displayName": "Variables",
+            "isExpanded": false,
+            "visibleRule": "command = validate || command = plan || command = apply"
+        }
+    ],
+    "inputs": [{
+            "name": "command",
+            "type": "pickList",
+            "label": "Command",
+            "defaultValue": "validate",
+            "required": true,
+            "helpMarkDown": "The terraform command to run.",
+            "options": {
+                "init": "init",
+                "validate": "validate",
+                "plan": "plan",
+                "apply": "apply",
+                "destroy": "destroy"
+            },
+            "properties": {
+                "EditableOptions": "False"
+            }
+        },
+        {
+            "name": "workingDirectory",
+            "type": "filePath",
+            "label": "Configuration Directory",
+            "defaultValue": "$(System.DefaultWorkingDirectory)",
+            "helpMarkDown": "Terraform configuration directory. If empty, the root of the repository (build) or artifacts (release) is used, which is $(System.DefaultWorkingDirectory)",
+            "required": "false"
+        },
+        {
+            "name": "environmentServiceName",
+            "type": "connectedService:AzureRM",
+            "label": "Environment Azure Subscription",
+            "required": false,
+            "visibleRule": "command = plan || command = apply",
```

@jakebair add destroy here as well
```suggestion
            "visibleRule": "command = plan || command = apply || command = destroy",
```